src/clj/game/core/abilities.clj

```diff
@@ -106,16 +106,16 @@
   :makes-run -- indicates if the ability makes a run."
   ;; perhaps the most important function in the game logic
   ([state side {:keys [eid] :as ability} card targets]
-   (resolve-ability state side (or eid (make-eid state)) ability card targets))
+   (resolve-ability state side (or eid (make-eid state {:source card :source-type :ability})) ability card targets))
   ([state side eid ability card targets]
-   (resolve-ability-eid state side (assoc ability :eid eid) card targets)))
+   (resolve-ability-eid state side (assoc ability :eid eid :source card :source-type :ability) card targets)))
```

Comment:
This is being `assoc`'d to `ability`, not `eid`. Is that on purpose?

Reply:
This was on purpose. If I'm reading `resolve-ability-eid` correctly, it expects the `eid` to be wrapped inside the `ability`. I found this weird too, but didn't want to change it, since I don't know what might have been the motivation behind this.